fvm/src/syscalls/crypto.rs

```diff
@@ -43,6 +43,28 @@ pub fn verify_signature(
         .map(|v| if v { 0 } else { -1 })
 }
 
+pub fn recover_public_key(
+    context: Context<'_, impl Kernel>,
+    hash_off: u32,
+    hash_len: u32,
+    sig_off: u32,
+    sig_len: u32,
+) -> Result<[u8; SECP_PUB_LEN]> {
```

Comment:
If these are actually fixed-length, we can just pass the offset and not the length.